pkg/metrics/metrics.go

```diff
@@ -103,7 +119,7 @@ The label 'result' can have the following values:
 		prometheus.HistogramOpts{
 			Subsystem: constants.KueueName,
 			Name:      "admission_wait_time_seconds",
-			Help:      "The time between a Workload was created until it was admitted, per 'cluster_queue'",
+			Help:      "The time from when a workload got the quota reservation until admission, per 'cluster_queue'",
```

Comment:
This is a slightly different approach than in the Issue description, but I'm ok with either.

However, with the change of semantics I suggest a rename, maybe: `quota_reserved_to_admission_wait_time_seconds`

Another approach would be to call it `admission_checks_wait_time_seconds`, but then do not count 0 if there are no admission checks.

I'm slightly leaning towards `quota_reserved_to_admission_wait_time_seconds`. WDYT @alculquicondor @tenzen-y ?

Reply:
Can I just rename metric to quota_reserved_to_admission_wait_time_seconds? Or should I need to change logic?

Reply:
`quota_reserved_to_admission_wait_time_seconds` sgtm. @alculquicondor any view?

Reply:
sgtm. I would also not count a 0 if there are no admission checks

Reply:
SGTM

> I would also not count a 0 if there are no admission checks

This is a good call. I agree with this.

Reply:
Then , maybe the name of the metric should be something like `admission_check_wait_time_seconds ` or keep two histograms. The transition from reserving to admitted happens with or without admissioncheckes .

In my opinion the way they are now, the counter and the number of recorded histogram records should be the same always.

Reply:
To follow up on the idea of `admission_check_wait_time_seconds` I opened: https://github.com/kubernetes-sigs/kueue/issues/1994.

Reply:
Can you summarize the full plan including all the involved metrics?

Reply:
I agree with everything, except counting cases with no admission checks as zero in point 5.

Someone that wants to actually monitor `quota_reserved_to_admission_wait_time_seconds` doesn't care about queues that don't have admission checks.

Reply:
I see, but the name of the metric is sort of agnostic about the presence of admission checks. 

What would you say about calling it `admission_checks_wait_time_seconds`, and only counting when there are admission checks?

Reply:
SGTM

Reply:
> 1. fix `admissionWaitTime` (`admission_wait_time_seconds`) - measures time to "Admitted" condition since creation time, or last eviction time

This is problematic the sense that when a quota reservation is done the eviction condition is altered and we are not able to recover that last evicted time, this is the reason why the event recording is done with the time from reservation to admission.

Reply:
Good point. So to implement this I see two ways:
1. have an extra condition like `Queued` with `LastTransitionTime` pointing to creation or last eviction, which is not updated on quota reservation, but on eviction
2. extra field in workload status to know how long quota reservation took, like `timeToReserveQuotaMs`

I would be leaning towards (1.) as more flexible.

Reply:
I like it